fdbserver/WorkerInterface.actor.h

```diff
@@ -865,16 +865,17 @@ ACTOR Future<Void> clusterController(Reference<ClusterConnectionFile> ccf,
 class IKeyValueStore;
 class ServerCoordinators;
 class IDiskQueue;
+struct MockLogSystem;
 ACTOR Future<Void> storageServer(
     IKeyValueStore* persistentData,
     StorageServerInterface ssi,
     Tag seedTag,
     ReplyPromise<InitializeStorageReply> recruitReply,
     Reference<AsyncVar<ServerDBInfo>> db,
     std::string folder,
-    // Only applicable when logSystemType is mock. This is actually a pointer to MockLogSystem but we cannot specify it
-    // because ILogSystem already depends on WorkerInterface.
-    void* mockLogSystem = nullptr);
+    // Only applicable when logSystemType is mock.
+    // This has to be a shared_ptr rather than unique_ptr or Reference because MockLogSystem is only forward declared.
+    std::shared_ptr<MockLogSystem> mockLogSystem = std::shared_ptr<MockLogSystem>());
```

Comment:
Does this need to be a member variable? Or could it just be constructed as needed when passed to `storageServerCore`?